src/group_theory/commensurable.lean

```diff
@@ -0,0 +1,118 @@
+/-
+Copyright (c) 2021 Chris Birkbeck. All rights reserved.
+Released under Apache 2.0 license as described in the file LICENSE.
+Authors: Chris Birkbeck
+-/
+import group_theory.index
+import group_theory.quotient_group
+import group_theory.subgroup.pointwise
+
+/-!
+# Commensurability for subgroups
+
+This file defines commensurability for subgroups of a group `G`. It then goes on to prove that
+commensurability defines an equivalence relation and finally defines the commensurator of a subgroup
+of `G`.
+
+## Main definitions
+
+* `commensurable`: defines commensurability for two subgroups `H`, `K` of  `G`
+* `commensurator`: defines the commensurator of a a subgroup `H` of `G`.
+-/
+
+variables {G : Type*} [group G]
+
+/--Two subgroups `H K` of `G` are commensurable if `H ⊓ K` has finite index in both `H` and `K` -/
+def commensurable (H K : subgroup G) : Prop :=
+H.relindex K ≠ 0 ∧ K.relindex H ≠ 0
+
+namespace commensurable
+
+@[refl] protected lemma refl (H : subgroup G) : commensurable H H := by
+ simp only [commensurable, nat.one_ne_zero, ne.def, not_false_iff, and_self, subgroup.relindex_self]
+
+lemma comm {H K : subgroup G} : commensurable H K ↔ commensurable K H := and.comm
+
+@[symm] lemma symm {H K : subgroup G} : commensurable H K → commensurable K H := and.symm
+
+@[trans] lemma trans {H K L : subgroup G} (hhk : commensurable H K ) (hkl : commensurable K L) :
+  commensurable H L :=
+⟨subgroup.relindex_ne_zero_trans H K L hhk.1 hkl.1,
+  subgroup.relindex_ne_zero_trans L K H hkl.2 hhk.2⟩
+
+lemma equivalence : equivalence (@commensurable G _) :=
+⟨commensurable.refl, λ _ _, commensurable.symm, λ _ _ _, commensurable.trans⟩
+
+/--Equivalence of `K/H ⊓ K` with `gKg⁻¹/gHg⁻¹ ⊓ gKg⁻¹`-/
+noncomputable def  quot_conj_equiv (H K : subgroup G) (g : G) :
+  quotient_group.quotient (H.subgroup_of K) ≃
+  quotient_group.quotient ((subgroup.conj_subgroup g H).subgroup_of (subgroup.conj_subgroup g K)) :=
```

Comment:
You will need to `import group_theory.group_action.conj_act` and `open_locale pointwise`.
```suggestion
noncomputable def  quot_conj_equiv (H K : subgroup G) (g : conj_act G) :
  quotient_group.quotient (H.subgroup_of K) ≃
  quotient_group.quotient ((g • H).subgroup_of (g • K)) :=
```

Reply:
Ok I've done it this way now. I removed the definition of `conj_subgroup` from the `pointwise` file.